src/density-qvm.lisp

```diff
@@ -32,55 +31,52 @@
         (not simple-array)))
 
 (defclass density-qvm (pure-state-qvm)
-  ((matrix-view :reader density-matrix-view
-                :type density-operator-matrix-view
-                :documentation "This is a 2D array displaced to the underlying QVM amplitudes.")
-   (temporary-state :accessor temporary-state
-                    :initform nil
-                    :documentation "Sometimes the simulation needs to use some temporary state. We save it so we don't have to alloc/dealloc frequently.")
+  ((state :accessor state
+          :initarg :state)
    (noisy-gate-definitions :initarg :noisy-gate-definitions
                            :accessor noisy-gate-definitions
-                           :initform (make-hash-table :test 'equalp)
-                           :documentation "Noisy gate definitions that, if present, override those stored in GATE-DEFINITIONS.")
+                           :initform (make-hash-table :test 'equalp))
    (readout-povms
     :initarg :readout-povms
     :accessor readout-povms
-    :initform (make-hash-table)
-    :documentation "Noisy readout encoded as diagonal single qubit
-    POVM given as a 4-element list
-
-          (p(0|0) p(0|1)
-           p(1|0) p(1|1))
-
-which for each qubit gives the probability p(j|k) of measuring outcome
-j given actual state k. Note that we model purely classical readout
-error, i.e., the post measurement qubit state is always k, but the
-recorded outcome j may be different."))
+    :initform (make-hash-table)))
   (:documentation "A density matrix simulator."))
 
 
 ;;; Creation and Initialization
 
+(defmethod amplitudes ((qvm density-qvm))
+  (amplitudes (state qvm)))
+
+(defmethod (setf amplitudes) (new-amplitudes qvm)
+  (setf (amplitudes (state qvm)) new-amplitudes))
+
+(defmethod temporary-state ((qvm density-qvm))
+  (temporary-state (state qvm)))
+
+(defmethod (setf temporary-state) (temp-storage (qvm density-qvm))
+  (setf (temporary-state (state qvm)) temp-storage))
+
+(defmethod density-matrix-view ((qvm density-qvm))
```

Comment:
I'm wondering if we should have this method defined on the `qvm` itself. Some tests use it, but otherwise what is gained vs lost in just using `(matrix-view (state qvm))` as opposed to `(density-matrix-view qvm)`?

Reply:
Similarly to the `temporary-state` methods defined on the qvms, I included this method when I got rid of the `density-matrix-view` accessor for this qvm.